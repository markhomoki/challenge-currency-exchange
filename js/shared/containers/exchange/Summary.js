import * as React from 'react';
import { connect } from 'react-redux';
import { Button } from 'app/components';
import { CURRENCIES } from 'app/types';

@connect((state) => {
	return {
		summary: state.summary,
	};
})
export default class Summary extends React.PureComponent {

	hide = () => {
		this.props.dispatch({
			type: 'SUMMARY_HIDE',
		});
	}

	renderSymbol = (cur) => {
		return CURRENCIES[cur] || '';
	}

	render() {
		const {
			summary,
		} = this.props;

		const hiddenClass = !summary.show ? 'hidden' : '';

		return (
			<div className={`summary ${hiddenClass}`}>
				<div className="summary-info-wrapper">
					<div className="complete-icon">
						<i className="ion-checkmark" />
					</div>
					<p className="info">You exchanged {this.renderSymbol(summary.fromCur)}{summary.fromVal.toFixed(2)} <i className="ion-android-arrow-forward" /> {this.renderSymbol(summary.toCur)}{summary.toVal.toFixed(2)}</p>
				</div>
				<div className="btn-wrapper">
					<Button onClick={this.hide}>Done</Button>
				</div>
			</div>
		);
	}

}
