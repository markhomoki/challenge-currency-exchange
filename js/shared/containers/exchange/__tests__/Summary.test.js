import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from 'app/store';
import Summary from '../Summary';

describe('Exchange.Summary', () => {

	let wrapper;
	let instance;

	beforeEach(() => {
		wrapper = shallow(<Summary store={store} />).dive();
		instance = wrapper.instance();
	});

	it('renders correctly', () => {
		const tree = renderer.create(<Provider store={store}><Summary /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('only renders when summary.show is true', () => {
		const shouldShow = instance.props.summary.show;

		expect(wrapper.find('.summary').hasClass('hidden')).toBe(!shouldShow);
	});

});
