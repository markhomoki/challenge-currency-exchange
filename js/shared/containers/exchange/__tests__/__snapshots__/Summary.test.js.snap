// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange.Summary renders correctly 1`] = `
<div
  className="summary hidden"
>
  <div
    className="summary-info-wrapper"
  >
    <div
      className="complete-icon"
    >
      <i
        className="ion-checkmark"
      />
    </div>
    <p
      className="info"
    >
      You exchanged 
      
      0.00
       
      <i
        className="ion-android-arrow-forward"
      />
       
      
      0.00
    </p>
  </div>
  <div
    className="btn-wrapper"
  >
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      Done
    </button>
  </div>
</div>
`;
